modules/ROOT/pages/cross_cutting/test_automation/sub_system_tests.adoc

```diff
@@ -0,0 +1,460 @@
+    }
+    @Override
+    public void afterAll(ExtensionContext extensionContext)  {
+        if (null != wireMockServer) {
+            wireMockServer.stop();
+        }
+    }
+    @Override
+    public void beforeEach(ExtensionContext extensionContext)  {
+        WireMock.resetAllRequests();
+        WireMock.stubFor(post(urlEqualTo(TestPatchs.myApi()))
+                .willReturn(aResponse()
+                    .withHeader("Content-Type", "application/json")
+                    .withStatus(200)));
+    }
+}
+
+--
+
+Example Junit test with Wiremock rule::
++
+--
+[source, java]
+@ExtendWith(LoggedInRequestSetup.class)
+@ExtendWith(WiremockSetup.class)
+public class BookingConfirmationApiTest {
+    @Test
+    public void bookingNotSuccessfulForNotWorkingEmail(RequestSpecification request) {
+        // prepare mock
+        final String errorEmail = "myveryspecialemailforerror@error.de";
+        WireMock.stubFor(post(urlEqualTo(TestPaths.myApi()))
+                .withRequestBody(matchingJsonPath("recipient", equalTo(errorEmail)))
+                .willReturn(aResponse().withStatus(400)));
+        // call system under test (implicitly calls "myApi") and verify result
+        Booking booking = BookingBuilder.defaultBooking().withEmail(errorEmail);
+        request.body(booking)
+                .when()
+                .post(TestPaths.createBooking())
+                .then().statusCode(500);
+        // verify that Wiremock is called during the test
+        List<LoggedRequest> requests = WireMock.findAll(
+            postRequestedFor(urlEqualTo(TestPaths.myApi())));
+        assertThat(requests).hasSize(1);
+        ...
+    }
+}
+
+--
+
+== UI-related Good Practices
+
+====
+*UI-test* in this section means a functional test that uses UI interaction to trigger or check functionality of a fully integrated system under test.
+This is a difference to Unit- or Unit-Integration Tests of a UI components.
+====
+
+Typical tool: https://www.selenium.dev/documentation/webdriver/[Selenium WebDriver]
+
+=== Encapsulate UI technology specific logic in page objects
+When the UI changes, you often need to change your UI tests.
+But if the change is just affecting the structure of one page, you don't want to change all tests that use the page.
+To prevent this, encapsulate the access to widgets on a page in a page object.
+More details can be found https://www.selenium.dev/documentation/test_practices/encouraged/page_object_models/[here].
+
+=== Encapsulate UI components in your test code
+Web development often works with libraries or patterns of re-usable UI components that take care for a unique look and feel. 
+They change over time.  
+Like the motivation for the page object pattern, you don't want to change all tests or all page objects, if the internal structure of such a component changes. 
+So, similar to page objects, create "widget objects" that encapsulate the access to common UI components.
```

Comment:
My understanding is that these 'widget objects' are then used in the 'page objects'. My be one additional sentence would make this a bit clearer

Reply:
done